jplag/src/main/java/de/jplag/JPlagResult.java

```diff
@@ -136,11 +136,14 @@ private int[] calculateSimilarityDistribution(List<JPlagComparison> comparisons)
 
     private int[] calculateDistributionFor(List<JPlagComparison> comparisons, Function<JPlagComparison, Float> similarityExtractor) {
         int[] similarityDistribution = new int[SIMILARITY_DISTRIBUTION_SIZE];
+        int similarityDistributionBucketSize = 100 / SIMILARITY_DISTRIBUTION_SIZE;
         for (JPlagComparison comparison : comparisons) {
-            Float similarity = similarityExtractor.apply(comparison);
-            int index = (int) (similarity / SIMILARITY_DISTRIBUTION_SIZE);
-            index = index == SIMILARITY_DISTRIBUTION_SIZE ? SIMILARITY_DISTRIBUTION_SIZE - 1 : index;
-            similarityDistribution[SIMILARITY_DISTRIBUTION_SIZE - 1 - index]++;
+            float similarity = similarityExtractor.apply(comparison); // extract similarity in percent: 0f <= similarity <= 100f
+            int index = (int) (similarity / similarityDistributionBucketSize); // divide similarity by bucket size to find index of correct bucket.
+            if (index == SIMILARITY_DISTRIBUTION_SIZE)
+                index--; // index is out of bounds when similarity is 100%. decrease by one to count towards the highest value bucket
```

Comment:
According to the comments:

Maybe `index = Math.min((int)(similarity / similarityDistributionBucketSize), SIMILARITY_DISTRIBUTION_SIZE-1)`

:)

Reply:
Is this resolved @nestabentum?